From 5cb7e07ced0ff3b3289f41536d58fbe506c5f65e Mon Sep 17 00:00:00 2001
From: Ferdinand Xu <cheng.a.xu@intel.com>
Date: Wed, 16 Sep 2015 22:07:47 -0400
Subject: [PATCH 0711/1164] CDH-42376 : HIVE-11796: CLI option is not updated
 when executing the initial
 files[beeline-cli](Ferdinand Xu, reviewed by
 Xuefu Zhang)

Change-Id: I8939a4465551e32b2102df725e149af8ab144748
---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |   24 +-------------------
 .../src/java/org/apache/hive/beeline/Commands.java |    4 ++++
 2 files changed, 5 insertions(+), 23 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index a9a4dd2..ef2147a 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -709,7 +709,6 @@ int initArgsFromCliVars(String[] args) {
     if (!commands.isEmpty()) {
       embeddedConnect();
       connectDBInEmbededMode();
-      updateOptsForCli();
       for (Iterator<String> i = commands.iterator(); i.hasNext(); ) {
         String command = i.next().toString();
         debug(loc("executing-command", command));
@@ -857,7 +856,7 @@ private String obtainPasswordFromFile(String passwordFilePath) {
     }
   }
 
-  private void updateOptsForCli() {
+  public void updateOptsForCli() {
     getOpts().updateBeeLineOptsFromConf();
     getOpts().setShowHeader(false);
     getOpts().setOutputFormat("dsv");
@@ -890,9 +889,6 @@ public int begin(String[] args, InputStream inputStream) throws IOException {
           return code;
         }
         defaultConnect(false);
-        updateOptsForCli();
-
-        processInitFiles(opts.getInitFiles());
       }
 
       if (getOpts().isHelpAsked()) {
@@ -983,24 +979,6 @@ private int executeFile(String fileName) {
     }
   }
 
-  /**
-   * Only initial files specified by i option will be executed. The hiverc file will be processed by session manager.
-   *
-   * @param files
-   * @throws IOException
-   */
-  public void processInitFiles(String[] files) throws IOException {
-    if (files == null || files.length == 0) {
-      return;
-    }
-    for (String initFile : files) {
-      int rc = executeFile(initFile);
-      if (rc != 0) {
-        System.exit(rc);
-      }
-    }
-  }
-
   private int execute(ConsoleReader reader, boolean exitOnError) {
     String line;
     while (!exit) {
diff --git a/beeline/src/java/org/apache/hive/beeline/Commands.java b/beeline/src/java/org/apache/hive/beeline/Commands.java
index b7bc05f..9063418 100644
--- a/beeline/src/java/org/apache/hive/beeline/Commands.java
+++ b/beeline/src/java/org/apache/hive/beeline/Commands.java
@@ -1473,6 +1473,10 @@ public boolean connect(Properties props) throws IOException {
       beeLine.getDatabaseConnections().setConnection(
           new DatabaseConnection(beeLine, driver, url, props));
       beeLine.getDatabaseConnection().getConnection();
+
+      if (!beeLine.isBeeLine()) {
+        beeLine.updateOptsForCli();
+      }
       beeLine.runInit();
 
       beeLine.setCompletions();
-- 
1.7.9.5

